Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/SingleFileTransfer.cs

```diff
@@ -295,7 +293,7 @@ private void ExecuteSourceOperationMoveOrRename()
             var destFileName = Path.Combine(moveToPath, SourceFile.Name);
 
             try { File.Move(filePath, destFileName); }
-            catch { throw new Exception($"Failure in source operation: File {Path.GetFileName(destFileName)} exists in move to directory."); }
+            catch (Exception ex) { throw new Exception($"Failure in source operation: {ex}: {ex.Message}"); }
```

Comment:
"{ex}" calles .ToString() and it usually already contains the ex.Message, so this exception text will have the exception message twice. You can replace the first {ex} with {ex.GetType().Name} so it is clear what type of exception is happening. Also, pass the original exception as inner exception like this: `new Exception("text", originalException);`

Reply:
Fixed.

Reply:
Ok, not the error message does not contain the `ex.Message`, which is fine,  but in this case the `GetType().Name` is also not very useful. Now the error message is like this: "Failure in source op: InvalidOperationException". This is not useful for anyone.

Depending on context you might want to include the original exception message in you own exception message. E.g. here, if we wanted to quote the original exception in our new exception we would do "Failure in source op: InvalidOperationException - division by zero not allowed". This is already closer to something understandable, so the actualy code would be `new Exception($"Failure in source operation: {ex.GetType().Name}: {ex.Message}", ex)`. This way your exception has some better and more useful information.

The above method of including original message in new exception is usually needed in cases where we at some point lose the inner exception for some reason. In case of Frends I believe that it can handle inner exceptions nicely, so I think that even a simple `new Exception("Failure in source op", ex)` would suffice. Because Frends can show both you text (which is a good added context for the reader) AND also it can show inner exceptions in UI.

Reply:
Fixed and updated tests.